holoviews/util.py

```diff
@@ -28,14 +28,17 @@ class Dynamic(param.ParameterizedFunction):
     kwargs = param.Dict(default={}, doc="""
         Keyword arguments passed to the function.""")
 
+    shared_data = param.Boolean(default=False, doc="""
+        Whether the cloned DynamicMap will share the same data.""")
```

Comment:
As ``.data`` for a ``DynamicMap`` is the callback, it would be good to explain what sharing/not sharing means here. I'm not quite sure what it means to be honest so it might be something we want to document in other places too.